vcomponent/src/vcomponent_hdmi_cec_device.c

```diff
@@ -0,0 +1,465 @@
+  {"APPLE", VENDOR_CODE_APPLE},
+  {"HARMAN_KARDON2", VENDOR_CODE_HARMAN_KARDON2},
+  {"GOOGLE", VENDOR_CODE_GOOGLE},
+  {"AKAI", VENDOR_CODE_AKAI},
+  {"AOC", VENDOR_CODE_AOC},
+  {"PANASONIC", VENDOR_CODE_PANASONIC},
+  {"PHILIPS", VENDOR_CODE_PHILIPS},
+  {"DAEWOO", VENDOR_CODE_DAEWOO},
+  {"YAMAHA", VENDOR_CODE_YAMAHA},
+  {"GRUNDIG", VENDOR_CODE_GRUNDIG},
+  {"PIONEER", VENDOR_CODE_PIONEER},
+  {"LG", VENDOR_CODE_LG},
+  {"SHARP", VENDOR_CODE_SHARP},
+  {"SONY", VENDOR_CODE_SONY},
+  {"BROADCOM", VENDOR_CODE_BROADCOM},
+  {"SHARP2", VENDOR_CODE_SHARP2},
+  {"VIZIO", VENDOR_CODE_VIZIO},
+  {"BENQ", VENDOR_CODE_BENQ},
+  {"HARMAN_KARDON", VENDOR_CODE_HARMAN_KARDON},
+  {"UNKNOWN", VENDOR_CODE_UNKNOWN},
+  };
+
+const static strVal_t gPSStrVal [] = {
+  { "on", (int)POWER_STATUS_ON  },
+  { "off", (int)POWER_STATUS_OFF  },
+  { "standby", (int)POWER_STATUS_STANDBY },
+  { "unknown", (int)POWER_STATUS_UNKNOWN }
+};
+
+/* Load the device info into the passed in vCHdmiCec_device_info_t*
+* prefix can be 
+*   "hdmicec/device_map/0"
+*   "hdmicec/device_map/0/children/0"
+*   "hdmicec/device_map/0/children/1"
+*   "hdmicec/device_map/0/children/1/children/0"
+*/
+void LoadDeviceInfo (ut_kvp_instance_t* instance, char* prefix, struct vCHdmiCec_device_info_t* device)
+{
+  char tmp[strlen(prefix) + 64];
+  char type[32];
+
+  if (device == NULL || instance == NULL || prefix == NULL)
+  {
+    assert(device != NULL);
+    assert(instance != NULL);
+    assert(prefix != NULL);
+    return;
+  }
+  strcpy(tmp, prefix);
+
+  strcpy(tmp + strlen(prefix), "/name");
+  ut_kvp_getStringField(instance, tmp, device->osd_name, MAX_OSD_NAME_LENGTH);
+
+  strcpy(tmp + strlen(prefix), "/active_source");
+  device->active_source = ut_kvp_getBoolField(instance, tmp);
+
+  strcpy(tmp + strlen(prefix), "/pwr_status");
+  ut_kvp_getStringField(instance, tmp, type, sizeof(type));
+  device->power_status = vCHdmiCec_GetValue(gPSStrVal, sizeof(gPSStrVal)/sizeof(strVal_t), type, (int)POWER_STATUS_UNKNOWN);
+
+  strcpy(tmp + strlen(prefix), "/version");
+  device->version = (vCHdmiCec_version_t) ut_kvp_getUInt32Field(instance, tmp);
+
+  strcpy(tmp + strlen(prefix), "/vendor");
+  ut_kvp_getStringField(instance, tmp, type, sizeof(type));
+  device->vendor_id = vCHdmiCec_GetValue(gVCStrVal, sizeof(gVCStrVal)/sizeof(strVal_t), type, (int)VENDOR_CODE_UNKNOWN);
+
+  strcpy(tmp + strlen(prefix), "/type");
+  ut_kvp_getStringField(instance, tmp, type, sizeof(type));
+  device->type = vCHdmiCec_GetValue(gDIStrVal, sizeof(gDIStrVal)/sizeof(strVal_t), type, (int)DEVICE_TYPE_UNKNOWN);
+
+  strcpy(tmp + strlen(prefix), "/port_id");
+  device->parent_port_id = ut_kvp_getUInt32Field(instance, tmp);
+
+  strcpy(tmp + strlen(prefix), "/number_children");
+  device->number_children = (vCHdmiCec_version_t) ut_kvp_getUInt32Field(instance, tmp);
+}
+
+struct vCHdmiCec_device_info_t* vCHdmiCec_Device_CreateMapFromProfile (ut_kvp_instance_t* instance, char* profile_prefix)
```

Comment:
you've for your naming & filenames different `vCHdmiCec_Device` & filename `vcomponent_hdmi_cec_device` you need to be consistant.